main_gui.py

```
@@ -0,0 +1,351 @@
+            
+            try:
+                # Create a temporary file that won't be deleted on close, so extract_audio can use it
+                with tempfile.NamedTemporaryFile(suffix=".wav", delete=False) as tmp_audio_file:
+                    temp_wav_path = tmp_audio_file.name
+                
+                self.processing_queue.put({'type': 'log', 'value': f"Temporary WAV path: {temp_wav_path}"})
+
+                if not extract_audio(video_path, temp_wav_path):
+                    # extract_audio should ideally raise an exception on failure or return more info
+                    self.processing_queue.put({'type': 'error', 'value': 'Audio extraction failed. The function returned false. Check console for details from audio_extractor.'})
+                    has_error = True; return # Stop processing
```

Ruff (0.11.9)

Multiple statements on one line (semicolon)

(E702)